### PR TITLE
Remake schedule page

### LINKED ISSUE #35: New style for the schedule page
Fix the styling on the schedule page, as well as the usability of past and future schedules.

### LINKED ISSUE #54: Schedule Page
Using the lectures api, request the user schedule and generate a custom page

### PR DESCRIPTION
**Pull request by ttoino**
Closes #35.

image

### CODE REVIEW

File: content-scripts/src/modules/schedule.js [hunk 1]
```diff
@@ -0,0 +1,437 @@
+import { injectOverrideFunctions } from "./initialize";
+import { lcmAll } from "./utilities/math";
+import removeElement from "./utilities/removeElement";
```
Comment (thePeras) [inline content-scripts/src/modules/schedule.js:3]:
```suggestion
import { lcmAll } from "./utilities/math";
```

Remove unused imports

**Review by thePeras — CHANGES_REQUESTED** [review]
Great work guys!!

I find a little catch in classes with type O, which I don't know what it means (online?).
There is no subtitle for that type in the currently Sigarra.
The class is "O"
Screenshot 2023-06-02 at 20 52 17

**Comment by ttoino** [comment]
> Great work guys!!
> 
> I find a little catch in classes with type O, which I don't know what it means (online?). There is no subtitle for that type in the currently Sigarra. The class is "O" Screenshot 2023-06-02 at 20 52 17

Whoops, I had never seen that class type before, nice catch! Looking at the course unit info, 'O' seems to stand for "Outra". @toni-santos which color shall we use for it? The default is gray.

**Comment by toni-santos** [comment]
> > Great work guys!!
> > I find a little catch in classes with type O, which I don't know what it means (online?). There is no subtitle for that type in the currently Sigarra. The class is "O" Screenshot 2023-06-02 at 20 52 17
> 
> Whoops, I had never seen that class type before, nice catch! Looking at the course unit info, 'O' seems to stand for "Outra". @toni-santos which color shall we use for it? The default is gray.

Seeing as this type of class in particular does not fit into any of the established types of classes (Practical, Theoretic, ...) and as such lacks the same semantic meaning there isn't much in terms of color coordination that we are able to do. With this being said, I believe using grey (or similar) would be the best option as it not only keeps with the color that users already know but it also does not take away emphasis from other color combinations we already established.

**Comment by toni-santos** [comment]
I have added colors for these classes, please check if it is working :)

**Comment by ttoino** [comment]
> I have added colors for these classes, please check if it is working :)

The selector was missing in the code, but it's working now

**Review by thePeras — CHANGES_REQUESTED** [review]
It appears that O is some kind of a special case. Its content is not inside a div.

Screenshot 2023-06-10 at 12 03 27
Screenshot 2023-06-10 at 12 04 46

**Comment by ttoino** [comment]
> It appears that O is some kind of a special case. Its content is not inside a div.
> 
> Screenshot 2023-06-10 at 12 03 27
> Screenshot 2023-06-10 at 12 04 46
> 

That's weird, that should have been fixed with my last commit, are you sure you pulled all the changes?

**Comment by thePeras** [comment]
> That's weird, that should have been fixed with my last commit, are you sure you pulled all the changes?

Sorry, its working. 
I notice the content-script weren't beeing rebuild.

Great work!!
✅ 🛫

**Review by thePeras — APPROVED** [review]

**Pull request merged**